src/app/hooks/useCustomEventTracker/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { use, useCallback } from 'react';
+import { VIEW_EVENT } from '#app/lib/analyticsUtils/analytics.const';
+import extractATITrackingProps from '#app/lib/analyticsUtils/extractATITrackingProps';
+import { sendEventBeacon } from '../../components/ATIAnalytics/beacon';
+import useTrackingToggle from '../useTrackingToggle';
+import { ServiceContext } from '../../contexts/ServiceContext';
+
+interface CustomEventData {
+  eventName: string;
+}
+
+/**
+ * A specialized React hook for tracking custom (non-click, non-view) events.
+ * Reverb is used to send the beacon. The event will appear in Piano under the "Event - Group" field.
+ * If a payload (`stringifiedData`) is provided to the `trackEvent` function, it will appear in Piano under the "Item name" field.
+ *
+ * @param {CustomEventData} eventName - A string representing the name of the custom event.
+ * @returns {Object} An object containing the `trackEvent` function, which can be called to trigger the event.
+ */
+
+const useCustomEventTracker = ({ eventName }: CustomEventData) => {
+  const {
+    pageIdentifier,
+    producerId,
+    platform,
+    statsDestination,
+    campaignID,
+    producerName,
+  } = extractATITrackingProps({
+    eventType: VIEW_EVENT,
+  });
+
+  const { trackingIsEnabled } = useTrackingToggle();
+  const { service, useReverb } = use(ServiceContext);
```

Comment:
This might be a question for @alex-magana, but do we still need `useReverb` since all services now do support reverb? (Aside - do we have a ticket for the removal of all useReverb references across Simorgh?)

Reply:
We no longer need `useReverb`. This will be removed from the codebase as part of [WS-222](https://bbc.atlassian.net/issues/WS-222) (previously [WORLDSERVICE-400](https://jira.dev.bbc.co.uk/browse/WORLDSERVICE-400)).